packages/cmf/src/cmfConnect.js

```diff
@@ -290,40 +285,27 @@ export default function cmfConnect({
 
 			onEventDispatch(props, key) {
 				if (CONSTANT.IS_HANDLER_DISPATCH_REGEX.test(key)) {
-					const handlerKey = key.replace(CONSTANT.IS_HANDLER_DISPATCH, '');
 					props.toOmit.push(key);
+					const handlerKey = key.replace(CONSTANT.IS_HANDLER_DISPATCH, '');
+					const original = props[handlerKey];
 					// eslint-disable-next-line no-param-reassign
-					props[handlerKey] = (event, data) => {
-						const payload = Object.assign(
-							{
-								event: serializeEvent(event),
-								data,
-							},
-							this.props[key],
-						);
-						this.props.dispatch(payload);
-						if (this.props[handlerKey]) {
-							this.props[handlerKey](event, data);
-						}
-					};
+					props[handlerKey] = onEvent.getOnEventDispatchHandler(this, this.props[key], original);
 				} else if (CONSTANT.IS_HANDLER_ACTION_CREATOR_REGEX.test(key)) {
 					props.toOmit.push(key);
 					const handlerKey = key.replace(CONSTANT.IS_HANDLER_ACTION_CREATOR, '');
-					let actionCreator = this.props[key];
-					if (typeof this.props[key] === 'object') {
-						actionCreator = this.props[key].id;
-					}
+					const original = props[handlerKey];
 					// eslint-disable-next-line no-param-reassign
-					props[handlerKey] = (event, data) => {
-						this.dispatchActionCreator(actionCreator, serializeEvent(event), {
-							props: this.props,
-							...data,
-							...(this.props[key].data || {}),
-						});
-						if (this.props[handlerKey]) {
-							this.props[handlerKey](event, data);
-						}
-					};
+					props[handlerKey] = onEvent.getOnEventActionCreatorHandler(
+						this,
+						this.props[key],
+						original,
+					);
+				} else if (CONSTANT.IS_HANDLER_SETSTATE_REGEX.test(key)) {
```

Comment:
We have here exactly 3 same patterns, with 3 variables : 
* the regex to test if we have an action creator, a dispatch, or a setState
* the constant string to remove from name, a suffix
* the getHandler() function

Perhaps we can factorize those duplications
```
onEventDispatch(props, key) {
	const eventConfiguration = getEventActionTypeConfiguration(key); // TODO
	if (eventConfiguration) {
		props.toOmit.push(key);
		const handlerKey = key.replace(eventConfiguration.SUFFIX, '');
		const originalHandler = props[handlerKey];
		// eslint-disable-next-line no-param-reassign
		props[handlerKey] = eventConfiguration.getHandler(this, this.props[key], originalHandler);
	}
}
```